mbuild/compound.py

```diff
@@ -324,6 +338,52 @@ def particles_by_element(self, element):
             if particle.element == element:
                 yield particle
 
+    @property
+    def mass(self):
+        """Return the total mass of a compound.
+
+        If the compound contains children compouds, the total mass of all
+        children compounds is returned.
+        If the compound contains element information (Compound.element) then
+        the mass is inferred from the elemental mass.
+        If Compound.mass has been set explicitly, then it will override the
+        mass inferred from Compound.element.
+        If neither of a Compound's element or mass attributes have been set,
+        then a mass of zero is returned.
+        """
+        if self._mass is not None:
+            return self._mass
+        else:
+            return sum([self._particle_mass(p) for p in self.particles()])
+
+    def _particle_mass(self, particle):
```

Comment:
This method should be static IMO. Because of hierarchical nature of this data-structure, we don't want an instance method here.

Reply:
tbh, I'm not super familiar with the the differences between static, class and instance methods, but let me know if the most recent change is what you had in mind.